### PR TITLE
Use Encoder/Decoder in RVD.persist

### PR DESCRIPTION
**Pull request by danking**
I also included a little change to use `Region.scoped` for writing a local `IndexedSeq[Annotation]`. In general, we cannot move Region off-heap until all the allocations of Regions are matched with `Region.close()`.

We don't use the context's region yet in the persist. I'll introduce that in a later pull request. This just avoids serializing regions and region values, which will soon become off-heap, non-serializable values.

### CODE REVIEW

**Comment by danking** [comment]
@cseed I shrunk the PR down to the minimal code change and it's still failing the `LDPruneSuite.testNoPrune` test. Do I misunderstand how to use the Encoder/Decorder stuff?

The failure isn't an assertion error. I've somehow changed the number of records that result from an LDPrune.

File: src/main/scala/is/hail/rvd/RVD.scala [hunk 1]
```diff
@@ -182,16 +184,16 @@ trait RVD {
...
-      val region = Region()
-      val rvb = new RegionValueBuilder(region)
       it.map { rv =>
-        region.clear()
-        rvb.start(localRowType)
-        rvb.addRegionValue(localRowType, rv)
-        val off = rvb.end()
-        RegionValue(region.copy(), off)
+        val baos = new ByteArrayOutputStream()
+        val enc = persistCodec.buildEncoder(baos)
```
Comment (cseed) [inline src/main/scala/is/hail/rvd/RVD.scala]: You should close baos.

File: src/main/scala/is/hail/rvd/RVD.scala [hunk 2]
```diff
@@ -201,9 +203,10 @@ trait RVD {
         .mapPartitions { it =>
           val region = Region()
           val rv2 = RegionValue(region)
-          it.map { rv =>
-            region.setFrom(rv.region)
-            rv2.setOffset(rv.offset)
+          it.map { bytes =>
+            val bais = new ByteArrayInputStream(bytes)
```
Comment (cseed) [inline src/main/scala/is/hail/rvd/RVD.scala]: Same.

File: src/main/scala/is/hail/rvd/RVD.scala [hunk 3]
```diff
@@ -201,9 +203,10 @@ trait RVD {
...
+            val dec = persistCodec.buildDecoder(bais)
+            rv2.setOffset(dec.readRegionValue(localRowType, region))
```
Comment (cseed) [inline src/main/scala/is/hail/rvd/RVD.scala]: You should reset region, `setFrom` had that effect.

File: src/main/scala/is/hail/rvd/RVD.scala [hunk 4]
```diff
@@ -182,16 +184,16 @@ trait RVD {
   protected def persistRVRDD(level: StorageLevel): PersistedRVRDD = {
     val localRowType = rowType
 
+    val persistCodec = CodecSpec.default
```
Comment (cseed) [inline src/main/scala/is/hail/rvd/RVD.scala]: This does a bit of compression which might not be quite what we want for cache().  Maybe make a new codec that just does Pack(Blocking(StreamBlock))?  (e.g. no lz4 or leb128)

**Pull request merged**